libraries/botbuilder-lg/src/templatesParser.ts

```diff
@@ -151,18 +167,21 @@ export class TemplatesParser {
      * @param importResolver Resolver to resolve LG import id to template text.
      * @param expressionParser Expression parser for evaluating expressions.
      * @param cachedTemplates Give the file path and templates to avoid parsing and to improve performance.
+     * @param parentTemplates Parent visited Templates.
      * @returns Entity.
      */
-    public static parseResource(
+    private static innerParseResource(
         resource: LGResource,
         importResolver?: ImportResolverDelegate,
         expressionParser?: ExpressionParser,
-        cachedTemplates?: Map<string, Templates>
+        cachedTemplates?: Map<string, Templates>,
+        parentTemplates?: Templates[]
     ): Templates {
         if (!resource) {
             throw new Error('lg resource is empty.');
         }
         cachedTemplates = cachedTemplates || new Map<string, Templates>();
+        parentTemplates = parentTemplates || [];
```

Comment:
Both of these can be expressed as defaults inside the parameter list.
```typescript
private static innerParseResource(..., cachedTemplates? = new Map<string, Templates>, parentTemplates?: Templates[] = [])
```

Reply:
Really learn a lot from you @joshgummersall . Big thanks!